### DISCUSSION TITLE
[Bug]: [OpenVino:] Intel compute runtime required for Frigate

### DISCUSSION THREAD

**Discussion by Redsandro** [blakeblackshear/frigate#12260] [Report a Bug] [answered]
### Describe the problem you are having

When starting from scratch:
- new hardware
- latest Proxmox 8.2
- latest Ubuntu 24.04 LXC
- latest Frigate 0.14.0-beta3

.. the OpenVino detector does not work with the Intel iGPU until the latest [Intel Compute Runtime](https://github.com/intel/compute-runtime/releases) is installed _inside the Frigate container_:

```bash
docker exec -ti frigate bash
cd $(mktemp -d)

wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.16900.23/intel-igc-core_1.0.16900.23_amd64.deb
wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.16900.23/intel-igc-opencl_1.0.16900.23_amd64.deb
wget https://github.com/intel/compute-runtime/releases/download/24.22.29735.20/intel-level-zero-gpu_1.3.29735.20_amd64.deb
wget https://github.com/intel/compute-runtime/releases/download/24.22.29735.20/intel-opencl-icd_24.22.29735.20_amd64.deb
wget https://github.com/intel/compute-runtime/releases/download/24.22.29735.20/libigdgmm12_22.3.19_amd64.deb

dpkg -i *.deb
```

I am aware that [none of the maintainers use OpenVino](https://github.com/blakeblackshear/frigate/discussions/10785#discussioncomment-9035425). The problem and the solution have been observed by [myself](https://github.com/blakeblackshear/frigate/discussions/12238) and at least one [other person](https://github.com/blakeblackshear/frigate/discussions/12012). 

If I understand correctly, Intel Compute is a dependency for OpenVino in Frigate, and it needs to be updated to the latest version in order to use the latest kernel.

I did a search and believe this is not being tracked as a bug yet. My apologies if I missed something.

### Steps to reproduce

1. Freshen your image (i.e. `docker compose down && docker compose up -d`)
2. Observe crash (e.g. `docker compose logs -f --tail=100`)
`[GPU] Context was not initialized for 0 device`
3. Change detector device to `CPU` in config
4. Bring up new config (e.g. `docker compose restart`)
5. Observe no crash
6. Install latest [Intel Compute Runtime](https://github.com/intel/compute-runtime/releases) (see above)
7. Change detector device back to `GPU` in config
8. Bring up new config (e.g. `docker compose restart`)
9. Observe no crash

### Version

0.14.0-beta3

### Frigate config file

```yaml
# https://docs.frigate.video/configuration/detectors
detectors:
  ov:
    type: openvino
    device: GPU
    model:
      path: /openvino-model/ssdlite_mobilenet_v2.xml

model:
  width: 300
  height: 300
  input_tensor: nhwc
  input_pixel_format: bgr
  labelmap_path: /openvino-model/coco_91cl_bkgr.txt
```

### Relevant log output

```shell
Process detector:ov:
Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/frigate/frigate/object_detection.py", line 102, in run_detector
    object_detector = LocalObjectDetector(detector_config=detector_config)
  File "/opt/frigate/frigate/object_detection.py", line 53, in __init__
    self.detect_api = create_detector(detector_config)
  File "/opt/frigate/frigate/detectors/__init__.py", line 18, in create_detector
    return api(detector_config)
  File "/opt/frigate/frigate/detectors/plugins/openvino.py", line 38, in __init__
    self.interpreter = self.ov_core.compile_model(
  File "/usr/local/lib/python3.9/dist-packages/openvino/runtime/ie_api.py", line 521, in compile_model
    super().compile_model(model, device_name, {} if config is None else config),
RuntimeError: Exception from src/inference/src/cpp/core.cpp:123:
Exception from src/inference/src/dev/plugin.cpp:59:
Check 'contexts.count(device_id)' failed at src/plugins/intel_gpu/src/plugin/plugin.cpp:204:
[GPU] Context was not initialized for 0 device
```

### Operating system

Proxmox

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

irrelevant

### Any other information that may be helpful

This is relevant for both Ubuntu 22.04 and 24.04.

Running `dpkg -i *.deb` above throws an error or two due to certain packages already being installed, but these can safely be ignored. [Here is a fix](https://github.com/blakeblackshear/frigate/discussions/10785#discussioncomment-9258468) for one package, but I didn't use it.

**Added:** I thought I was creating an `issue` but the issue wizard redirected this back to `discussions`. 
At least it has the `bug` label now. :slightly_smiling_face:

**Comment by NickM-27** [comment, marked as answer]
Thanks, this is a confirmed issue, created bug from the discussion

**Reply by Redsandro** [reply]
Can I track this somewhere so I know when it's safe to enable auto update?

**Reply by johnmmcgee** [reply]
> Can I track this somewhere so I know when it's safe to enable auto update?

https://github.com/blakeblackshear/frigate/issues/12266
